Data/ByteString.hs

```diff
@@ -889,6 +891,22 @@ take n ps@(BS x l)
     | otherwise = BS x n
 {-# INLINE take #-}
 
+-- | /O(1)/ 'takeEnd' @n xs@ is equivalent to @'drop' ('length' xs - n) xs@.
+-- Takes 'n' elements from end of bytestring.
+--
+-- >>> takeEnd 3 $ pack (c2w <$> "abcdefg")
+-- "efg"
+-- >>> takeEnd 0 $ pack (c2w <$> "abcdefg")
+-- ""
+-- >>> takeEnd 4 $ pack (c2w <$> "abc")
+-- "abc"
+takeEnd :: Int -> ByteString -> ByteString
+takeEnd n ps@(BS x len)
+  | (len - n) <= 0   = ps
+  | (len - n) >= len = empty
```

Comment:
Almost there! :)
Could you please simplify conditions in lines 905-906 to `n >= len` and `n <= 0`?

Reply:
I'd initially tried that but it makes the tests fail. 
When using `BS.take (BS.length str - n) str` the first value passed into `take` is `(BS.length str - n)` so because we inlined `take` we still need to negate `n` from `len`.

I'm fairly sure that's right?